woocommerce/woocommerce
# Only initializing background removal with jetpack connection present

### Changes proposed in this Pull Request:



This primary purpose of this is to ensure that the background removal feature is not initialized when no Jetpack connection is present. I also took the opportunity to refactor the client side index file, before it becomes a dumping ground for scattered initializations. 

Closes #40917 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch and activate on Jetpack-enabled store.
2. Ensure that you can see all features buttons, including title suggestions, product description, short product description, category suggestions and background removal. This ensure they're all initializing as they should.
3. Activate on non-jetpack connected store. All of the buttons for these features should not appear, with no errors present.
